Only in orinoco-0.13e-SN-15.orig: .orinoco.o.d
Common subdirectories: orinoco-0.13e-SN-15.orig/.tmp_versions and orinoco-0.13e-SN-15/.tmp_versions
diff -u orinoco-0.13e-SN-15.orig/hermes.h orinoco-0.13e-SN-15/hermes.h
--- orinoco-0.13e-SN-15.orig/hermes.h	2007-05-10 12:30:39.000000000 +0200
+++ orinoco-0.13e-SN-15/hermes.h	2007-05-10 12:36:58.000000000 +0200
@@ -206,7 +206,7 @@
 	u8 rate;
 	u8 rxflow;
 	u32 reserved;
-} __attribute__ ((packed));
+} ;
 
 #define HERMES_RXSTAT_ERR		(0x0003)
 #define	HERMES_RXSTAT_BADCRC		(0x0001)
@@ -226,7 +226,7 @@
 	u8 retry_count;
 	u8 tx_rate;
 	__le16 tx_control;	
-} __attribute__ ((packed));
+} ;
 
 #define HERMES_TXSTAT_RETRYERR		(0x0001)
 #define HERMES_TXSTAT_AGEDERR		(0x0002)
@@ -269,7 +269,7 @@
 	/* Those last are probably not available in very old firmwares */
 	__le16 RxDiscards_WEPICVError;
 	__le16 RxDiscards_WEPExcluded;
-} __attribute__ ((packed));
+} ;
 
 /* Grabbed from wlan-ng - Thanks Mark... - Jean II
  * This is the result of a scan inquiry command */
@@ -286,13 +286,13 @@
 	u8 rates[10];		/* Bit rate supported */
 	__le16 proberesp_rate;	/* Data rate of the response frame */
 	__le16 atim;		/* ATIM window time, Kus (hostscan only) */
-} __attribute__ ((packed));
+} ;
 /* Container */
 struct prism2_scan_frame {
 	__le16 rsvd;                   /* ??? */
 	__le16 scanreason;             /* ??? */
 	struct prism2_scan_apinfo aps[HERMES_SCANRESULT_MAX];        /* Scan result  : using the constant here, in case we change some day -- PlasmaHH*/
-} __attribute__ ((packed));
+} ;
 
 /* Same stuff for the Lucent/Agere card.
  * Thanks to h1kari <h1kari AT dachb0den.com> - Jean II */
@@ -306,7 +306,7 @@
 	/* bits: 0-ess, 1-ibss, 4-privacy [wep] */
 	__le16 essid_len;	/* ESSID length */
 	u8 essid[32];		/* ESSID of the network */
-} __attribute__ ((packed));
+} ;
 
 /* Moustafa: Scan structure for Symbol cards */
 struct symbol_scan_apinfo {
@@ -324,7 +324,7 @@
 	__le16 basic_rates;	/* Basic rates bitmask */
 	u8 unknown2[6];		/* Always FF:FF:FF:FF:00:00 */
 	u8 unknown3[8];		/* Always 0, appeared in f/w 3.91-68 */
-} __attribute__ ((packed));
+} ;
 
 union hermes_scan_info {
 	struct agere_scan_apinfo	a;
@@ -342,7 +342,7 @@
   
 struct hermes_linkstatus {
 	__le16 linkstatus;         /* Link status */
-} __attribute__ ((packed));
+} ;
 
 struct hermes_response {
 	u16 status, resp0, resp1, resp2;
@@ -352,11 +352,11 @@
 struct hermes_idstring {
 	u16 len;
 	u16 val[16];
-} __attribute__ ((packed));
+} ;
 
 typedef struct hermes_multicast {
 	u8 addr[HERMES_MAX_MULTICAST][ETH_ALEN];
-} __attribute__ ((packed)) hermes_multicast_t;
+}  hermes_multicast_t;
 
 
 // #define HERMES_DEBUG_BUFFER 1
diff -u orinoco-0.13e-SN-15.orig/ieee802_11.h orinoco-0.13e-SN-15/ieee802_11.h
--- orinoco-0.13e-SN-15.orig/ieee802_11.h	2007-05-10 12:30:39.000000000 +0200
+++ orinoco-0.13e-SN-15/ieee802_11.h	2007-05-10 12:36:58.000000000 +0200
@@ -22,7 +22,7 @@
 	u8 addr3[ETH_ALEN];
 	u16 seq_ctl;
 	u8 addr4[ETH_ALEN];
-} __attribute__ ((packed));
+} ;
 
 /* Frame control field constants */
 #define IEEE802_11_FCTL_VERS		0x0002
diff -u orinoco-0.13e-SN-15.orig/orinoco.c orinoco-0.13e-SN-15/orinoco.c
--- orinoco-0.13e-SN-15.orig/orinoco.c	2007-05-10 12:30:39.000000000 +0200
+++ orinoco-0.13e-SN-15/orinoco.c	2007-05-10 12:37:52.000000000 +0200
@@ -229,7 +229,7 @@
 	/* SNAP */
 	u8 oui[3];
 	u16 ethertype;
-} __attribute__ ((packed));
+} ;
 
 
 #define ENCAPS_OVERHEAD		(sizeof(encaps_hdr) + 2)
@@ -1296,7 +1296,7 @@
 	struct {
 		u16 len;
 		u16 type;
-	} __attribute__ ((packed)) info;
+	}  info;
 	int len, type;
 	int err;
 
@@ -1733,7 +1733,7 @@
 
 struct sta_id {
 	u16 id, variant, major, minor;
-} __attribute__ ((packed));
+} ;
 
 static int determine_firmware_type(struct net_device *dev, struct sta_id *sta_id)/*{{{*/
 {
@@ -2110,7 +2110,7 @@
 	} else {
 		struct {
 			u16 qual, signal, noise;
-		} __attribute__ ((packed)) cq;
+		}  cq;
 
 		err = HERMES_READ_RECORD(hw, USER_BAP,
 					 HERMES_RID_COMMSQUALITY, &cq);
@@ -4227,8 +4227,8 @@
 	strncpy(info->driver, DRIVER_NAME, sizeof(info->driver) - 1);
 	strncpy(info->version, DRIVER_VERSION, sizeof(info->version) - 1);
 	strncpy(info->fw_version, priv->fw_name, sizeof(info->fw_version) - 1);
-	if (dev->class_dev.dev)
-		strncpy(info->bus_info, dev->class_dev.dev->bus_id,
+	if (dev->dev.parent)
+		strncpy(info->bus_info, dev->dev.parent->bus_id,
 			sizeof(info->bus_info) - 1);
 	else
 		snprintf(info->bus_info, sizeof(info->bus_info) - 1,
@@ -4471,7 +4471,7 @@
 	priv->hw_unavailable = 1; /* orinoco_init() must clear this
 				   * before anything else touches the
 				   * hardware */
-	INIT_WORK(&priv->reset_work, (void (*)(void *))orinoco_reset, dev);
+	INIT_WORK(&priv->reset_work, dev);
 
 	priv->last_linkstatus = 0xffff;
 	priv->connected = 0;
diff -u orinoco-0.13e-SN-15.orig/orinoco.h orinoco-0.13e-SN-15/orinoco.h
--- orinoco-0.13e-SN-15.orig/orinoco.h	2007-05-10 12:30:39.000000000 +0200
+++ orinoco-0.13e-SN-15/orinoco.h	2007-05-10 12:36:58.000000000 +0200
@@ -31,7 +31,7 @@
 struct orinoco_key {
 	__le16 len;	/* always stored as little-endian */
 	char data[ORINOCO_MAX_KEY_SIZE];
-} __attribute__ ((packed));
+} ;
 
 typedef enum {
 	FIRMWARE_TYPE_AGERE,
@@ -45,18 +45,18 @@
 /* message data item for INT, BOUNDEDINT, ENUMINT */
 typedef struct p80211item_uint32
 {
-	uint32_t		did		__attribute__ ((packed));
-	uint16_t		status	__attribute__ ((packed));
-	uint16_t		len		__attribute__ ((packed));
-	uint32_t		data	__attribute__ ((packed));
-} __attribute__ ((packed)) p80211item_uint32_t;
+	uint32_t		did		;
+	uint16_t		status	;
+	uint16_t		len		;
+	uint32_t		data	;
+}  p80211item_uint32_t;
 
 typedef struct p80211msg
 {
-	uint32_t	msgcode		__attribute__ ((packed));
-	uint32_t	msglen		__attribute__ ((packed));
-	uint8_t	devname[WLAN_DEVNAMELEN_MAX]	__attribute__ ((packed));
-} __attribute__ ((packed)) p80211msg_t;
+	uint32_t	msgcode		;
+	uint32_t	msglen		;
+	uint8_t	devname[WLAN_DEVNAMELEN_MAX]	;
+}  p80211msg_t;
 
 #define DIDmsg_lnxind_wlansniffrm 0x0041
 #define DIDmsg_lnxind_wlansniffrm_hosttime 0x1041
@@ -85,7 +85,7 @@
 	p80211item_uint32_t	rate;
 	p80211item_uint32_t	istx;
 	p80211item_uint32_t	frmlen;
-} __attribute__ ((packed)) p80211msg_lnxind_wlansniffrm_t;
+}  p80211msg_lnxind_wlansniffrm_t;
 
 #define P80211ENUM_truth_false			0
 #define P80211ENUM_truth_true			1
@@ -186,12 +186,12 @@
 
 typedef struct p80211ioctl_req
 {
-	char 	name[WLAN_DEVNAMELEN_MAX] __attribute__ ((packed));
-	void	*data 		__attribute__ ((packed));
-	uint32_t	magic 	__attribute__ ((packed));
-	uint16_t	len 	__attribute__ ((packed));
-	uint32_t	result 	__attribute__ ((packed));
-} __attribute__ ((packed)) p80211ioctl_req_t;
+	char 	name[WLAN_DEVNAMELEN_MAX] ;
+	void	*data 		;
+	uint32_t	magic 	;
+	uint16_t	len 	;
+	uint32_t	result 	;
+}  p80211ioctl_req_t;
 
 struct orinoco_private {
 	void *card;	/* Pointer to card dependent structure */
diff -u orinoco-0.13e-SN-15.orig/orinoco_cs.c orinoco-0.13e-SN-15/orinoco_cs.c
--- orinoco-0.13e-SN-15.orig/orinoco_cs.c	2007-05-10 12:30:39.000000000 +0200
+++ orinoco-0.13e-SN-15/orinoco_cs.c	2007-05-10 12:36:58.000000000 +0200
@@ -431,7 +431,7 @@
 
 	/* Finally, report what we've done */
 	printk(KERN_DEBUG "%s: " DRIVER_NAME " at %s, irq %d, io "
-	       "0x%04x-0x%04x\n", dev->name, dev->class_dev.dev->bus_id,
+	       "0x%04x-0x%04x\n", dev->name, dev->dev.parent->bus_id,
 	       link->irq.AssignedIRQ, link->io.BasePort1,
 	       link->io.BasePort1 + link->io.NumPorts1 - 1);
 
